CHANGELOG.md

```diff
@@ -0,0 +1,10 @@
+# Changelog
```

Comment:
So far, pytometry managed its changelog using the [GitHub releases](https://github.com/scverse/pytometry/releases) page. I'm fine with either solution, but we should agree on one to avoid scattered/redundant information.

Reply:
Right, I also like how it's done right now! I'll remove the CHANGELOG file